userspace/libscap/scap_procs.c

```diff
@@ -557,6 +557,11 @@ static int32_t scap_proc_add_from_proc(scap_t* handle, uint32_t tid, int parentt
 	}
 
 	//
+	// Gathers the exepath
+	//
+	snprintf(tinfo->exepath, target_res, "%s", filename);
```

Comment:
My points were two:

1) If you use `target_res`, that's completely unrelated with the real size of `exepath`, so you are relying on the assumption that `sizeof(target_name) < sizeof(tinfo->exepath)`, if that isn't true you are then writing past the size of the buffer.

2) As I was also saying, you are really copying `filename` inside `exepath`, and I was pointing out that the string just contains the link to the exe in proc, for example `/proc/123/exe`, which is most likely not what you want, right?